### PR TITLE
feat(recall): #MZI-210 merge calendar logic in worker abstraction

### PR DESCRIPTION
**Pull request by aymericdevriesere**
## Describe your changes
Calendar worker class was written before abstract worker class, so this pr aims to include the calendar worker logic into the queueWorker logic. Also, some answers to event bus were missing in the ical worker, so I added them.
## Issue ticket number and link
[ MZI-210 ]
https://jira.support-ent.fr/browse/MZI-210
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: zimbra/src/main/java/fr/openent/zimbra/tasks/service/impl/data/SqlICalTaskService.java [hunk 1]
```diff
@@ -78,6 +78,11 @@ public Future<JsonObject> createTask(Action<ICalTask> action, ICalTask task) {
         return promise.future();
     }
 
+    @Override
+    protected Future<Void> createLogsForTask(ICalTask task, String error) {
+        return null;
```
Comment (alicedraillard) [inline zimbra/src/main/java/fr/openent/zimbra/tasks/service/impl/data/SqlICalTaskService.java:83]: pk null ?

File: zimbra/src/main/java/fr/openent/zimbra/worker/ICalRequestWorker.java [hunk 2]
```diff
@@ -51,148 +39,72 @@ public void start() throws Exception {
...
-
-        UserUtils.getUserInfos(this.eb, userId, user -> {
-            calendarService.getICal(user)
-                .compose(ical -> sendICalToCalendarModule(ical, task))
-                .onSuccess(result -> {
+    private void retrieveIcalAndNotifyCalendar(UserInfos user, ICalTask task) {
+        calendarService.getICal(user)
+                .onSuccess(ical -> {
                     queueService.editTaskStatus(task, TaskStatus.FINISHED)
                             .onFailure(err -> {
```
Comment (alicedraillard) [inline zimbra/src/main/java/fr/openent/zimbra/worker/ICalRequestWorker.java:71]: indentation ?

File: zimbra/src/main/java/fr/openent/zimbra/worker/ICalRequestWorker.java [hunk 3]
```diff
@@ -51,148 +39,72 @@ public void start() throws Exception {
...
-                                String errMessage = String.format("[Zimbra@%s::getICal]: error task status change: %s",
+                                String errMessage = String.format("[Zimbra@%s::retrieveIcalAndNotifyCalendar]: error task status change: %s",
+                                        this.getClass().getSimpleName(), err.getMessage());
+                                log.error(errMessage);
+                            });
+                    sendICalToCalendarModule(ical, task)
```
Comment (alicedraillard) [inline zimbra/src/main/java/fr/openent/zimbra/worker/ICalRequestWorker.java]: add parametre optionnel (eg sendICalToCalendarModule(ical, task, (Boolean) isFailed) --> if(isFailed) on change le icalMessage au lieu de faire 2 méthodes privées

File: zimbra/src/main/java/fr/openent/zimbra/model/task/Task.java [hunk 4]
```diff
@@ -2,14 +2,14 @@
 
 import fr.openent.zimbra.core.constants.Field;
 import fr.openent.zimbra.core.enums.TaskStatus;
+import fr.openent.zimbra.model.IModel;
```
Comment (qperie) [inline zimbra/src/main/java/fr/openent/zimbra/model/task/Task.java]: T'as pas l'air de l'utiliser

File: zimbra/src/main/java/fr/openent/zimbra/tasks/service/impl/data/SqlRecallTaskService.java [hunk 5]
```diff
@@ -68,6 +68,11 @@ protected Future<JsonObject> createTask(Action<RecallTask> action, RecallTask ta
         return promise.future();
     }
 
+    @Override
+    protected Future<Void> createLogsForTask(RecallTask task, String error) {
+        return null;
```
Comment (qperie) [inline zimbra/src/main/java/fr/openent/zimbra/tasks/service/impl/data/SqlRecallTaskService.java:73]: Même question qu'Alice plus haut

File: zimbra/src/main/java/fr/openent/zimbra/worker/ICalRequestWorker.java (hunk 2 above)
Comment (qperie) [reply zimbra/src/main/java/fr/openent/zimbra/worker/ICalRequestWorker.java:71]: Pas sût vu que c'est le .onFailure du editTaskStatus

File: zimbra/src/main/java/fr/openent/zimbra/worker/ICalRequestWorker.java [hunk 6]
```diff
@@ -51,148 +39,72 @@ public void start() throws Exception {
...
                             });
                 });
-        });
+    }
+
+    private Future<Void> notifyFailToCalendarModule(String error) {
+        Promise<Void> promise = Promise.promise();
+
+        JsonObject icalMessage = new JsonObject()
+                .put(Field.ACTION, "zimbra-platform-ics")
```
Comment (qperie) [inline zimbra/src/main/java/fr/openent/zimbra/worker/ICalRequestWorker.java]: Magic string ?

**Review by JinDouang — COMMENTED** [review]
cf comment alice et qper

File: zimbra/src/main/java/fr/openent/zimbra/tasks/service/impl/data/SqlICalTaskService.java (hunk 1 above)
Comment (aymericdevriesere) [reply zimbra/src/main/java/fr/openent/zimbra/tasks/service/impl/data/SqlICalTaskService.java:83]: Parceque la table n existe pas encore et du coup la requête sera dans la prochaine PR

File: zimbra/src/main/java/fr/openent/zimbra/tasks/service/impl/data/SqlRecallTaskService.java (hunk 5 above)
Comment (aymericdevriesere) [reply zimbra/src/main/java/fr/openent/zimbra/tasks/service/impl/data/SqlRecallTaskService.java:73]: Parceque la table n existe pas encore et du coup la requête sera dans la prochaine PR

**Review by qperie — APPROVED** [review]

File: zimbra/src/main/java/fr/openent/zimbra/worker/ICalRequestWorker.java (hunk 3 above)
Comment (alicedraillard) [reply zimbra/src/main/java/fr/openent/zimbra/worker/ICalRequestWorker.java]: @aymericdevriesere cf mon commentaire précédent, ça sert à rien de faire 2 méthodes différentes pr envoyer un objet quasi identique à la meme adresse (vu avec @qperie )

File: zimbra/src/main/java/fr/openent/zimbra/tasks/helpers/CalendarEventBusHelper.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package fr.openent.zimbra.tasks.helpers;
+
+import fr.openent.zimbra.core.constants.Field;
+import io.vertx.core.json.JsonObject;
+
+public class CalendarEventBusHelper {
+    public static JsonObject generateSuccessNotification(JsonObject result) {
```
Comment (alicedraillard) [inline zimbra/src/main/java/fr/openent/zimbra/tasks/helpers/CalendarEventBusHelper.java:7]: manque des parametres dans "result" pour le cas succeed :
String ical = data.getString(Field.ICS, "");
 String userId = data.getString(Field.USERID, "");
String platform = data.getString(Field.PLATFORM, "");
cf la [doc](https://confluence.support-ent.fr/display/MAgenda/Event+Bus)

**Review by alicedraillard — APPROVED** [review]

**Pull request merged**